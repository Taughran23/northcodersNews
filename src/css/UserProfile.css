.user-card {
    background-color: white;
    margin-left: 75px;
    margin-right: 75px;
    margin-top: 50px;
    padding: 15px;
    height: auto;
    overflow: auto;
}
.ArticleList {
    margin-left: 75px;
    margin-right: 75px;
    margin-top: 50px
}


@media screen and (max-width: 1000px) {
    .user-card {
        margin-left: 0px;
        margin-right: 0px
    }
      .ArticleList {
        margin-left: 0px;
        margin-right: 0px;
        margin-top: 50px;
        margin-bottom: 10px
    }
}

.avatar-container {
    float: right;
    height: auto
}