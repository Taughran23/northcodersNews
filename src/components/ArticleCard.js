import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

import '../css/ArticleCard.css';

export const ArticleCard = function (props) {
  return (
    <div className='article-container'>
      <article className='media'>
        <div className='media-left'>
          <div className='votes'>
            <div className='vote-up'>
              <i className='fa fa-chevron-up'onClick={props.articleVote.bind(null, props.id, 'up')}/>
            </div>
            <div className='vote-tally'>votes  {props.votes}</div>
            <div className='vote-down'>
              <i className='fa fa-chevron-down 'onClick={props.articleVote.bind(null, props.id, 'down')}/>
            </div>
          </div>
        </div>

        <div>
          <div>
            <NavLink to={`/articles/${props.id}`} className='article-title is-3'>{props.title}</NavLink>
          </div>
          <div>
            <NavLink to={`/users/${props.user}`} className='user-name is-4'>
              submitted by {props.user}
            </NavLink>
          </div>
        </div>
      </article>
    </div>
  );
};

ArticleCard.propTypes = {
  articleVote: PropTypes.func.isRequired,
  id: PropTypes.number.isRequired,
  votes: PropTypes.number.isRequired, 
  title: PropTypes.string.isRequired,
  user: PropTypes.string.isRequired
};

export default ArticleCard;