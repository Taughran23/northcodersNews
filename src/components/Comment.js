import React from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import commentDelete from '../actions/comment';
import { USERNAME } from '../config';

import '../css/Comment.css';


export class Comment extends React.Component {
  render () {
    return (
      <section>
        <div className='comment'>
          <NavLink className='author' to={`/users/${this.props.author}`}> {this.props.author}</NavLink>
          <div className='comment-body'>{this.props.body}</div>
          <div className='vote'>
            <button onClick={this.props.commentVote.bind(null, this.props.id, 'up', this.props.comments.data)}>Up</button>
            <div>Votes: {this.props.votes}</div>
            <button onClick={this.props.commentVote.bind(null, this.props.id, 'down',this.props.comments.data)}>Down</button>
          </div>
          {(function (props) {
            if (USERNAME === props.author) {
              return <button onClick={props.commentDelete.bind(null, props.id)}>
                          Delete Comment
              </button>;
            }
          })(this.props)}
        </div>
      </section>
    );
  }
}

function mapDispatchToProps (dispatch) {
  return {
    commentDelete: (id) => {
      dispatch(commentDelete(id));
    }
  };
}

function mapStateToProps (state) {
  return {
    comments: state.comments, 
    article: state.article
  };
}

Comment.propTypes = {
  id: PropTypes.string.isRequired,
  commentVote: PropTypes.func.isRequired, 
  votes: PropTypes.number.isRequired,
  body: PropTypes.string.isRequired, 
  author: PropTypes.string.isRequired, 
  comments: PropTypes.object.isRequired,

};

export default connect(mapStateToProps, mapDispatchToProps)(Comment);
