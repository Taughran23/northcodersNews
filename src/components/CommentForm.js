import React from 'react';
import PropTypes from 'prop-types';

import '../css/CommentForm.css';

const CommentForm = (props) => (
  <div className='comment-form'>
    <form className='form' onSubmit={props.submitHandler}>
      <textarea
        className="textarea"
        placeholder="Post comment"
        value={props.input}
        onChange={props.inputHandler}
      />
      <button
        className="button"
        type="submit"
        value="post"
      >
      Post comment
      </button>
    </form>

  </div>   
);

CommentForm.propTypes = {
  input: PropTypes.string.isRequired,
  submitHandler: PropTypes.func.isRequired, 
  inputHandler: PropTypes.func.isRequired
};

export default CommentForm;