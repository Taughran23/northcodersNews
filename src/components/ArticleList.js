import React from 'react';
import { connect  } from 'react-redux';
import PropTypes from 'prop-types';
import fetchArticles from '../actions/articles';
import * as actions from '../actions/articles';
import ArticleCard from './ArticleCard';

export class ArticleList extends React.Component {

  componentDidMount () {
    this.props.fetchArticles();
  }

  render () {
    return (
      <div>
        {this.props.articles.sort(function (a,b) { return b.votes - a.votes;})
          .map(article => 
            <ArticleCard 
              articleVote={this.props.articleVote}
              title={article.title} 
              votes={article.votes} 
              id={article._id}
              key={article.title} 
              user={article.created_by}/>)}
      </div>
    );
  }
}

function mapDispatchToProps (dispatch) {
  return {
    fetchArticles: () => {
      dispatch(fetchArticles());
    }, 
    articleVote:(id, vote) => {
      dispatch(actions.articleVote(id, vote));
    }
  };
}

function mapStateToProps (state) {
  return {
    articles: state.articles.data
  };
}

ArticleList.propTypes = {
  articles: PropTypes.array.isRequired,
  fetchArticles: PropTypes.func.isRequired,
  articleVote: PropTypes.func.isRequired
};

export default connect(mapStateToProps, mapDispatchToProps)(ArticleList);