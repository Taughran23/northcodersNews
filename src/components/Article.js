import React from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

import fetchArticle from '../actions/article';
import fetchComments from '../actions/comment';
import addComment from '../actions/addComment';
import commentVote from '../actions/commentVote';
import Comment from './Comment';
import CommentForm from './CommentForm';

import '../css/Article.css';

export class Article extends React.Component {
  constructor (props) {
    super(props);
    this.state = {
      commentTextInput: ''
    };
    this.inputHandler = this.inputHandler.bind(this);
    this.submitHandler = this.submitHandler.bind(this);
  }
  inputHandler(e) {
    this.setState({
      commentTextInput: e.target.value,
    });
  }

  submitHandler(e) {
    e.preventDefault();
    this.props.addComment(this.props.match.params.article_id, this.state.commentTextInput);
    this.setState({ commentTextInput: '' });
  }


  componentDidMount() {
    const article_id = this.props.match.params.article_id;
    this.props.fetchArticle (article_id);
    this.props.fetchComments(article_id);
  }

  render () {
    return (
      <div id='Article'>
        <div className="is-8">
          <div className="article">
            <h1 className="title"><b>{this.props.article.title}</b></h1>
            <div className='author'>
              <NavLink to={`/users/${this.props.article.created_by}`}className="subtitle author"> {this.props.article.created_by}</NavLink>
            </div>
            <p className="">{this.props.article.body}</p>
          </div>
        </div>
        <div className='comment'>
          <CommentForm 
            input={this.state.commentTextInput} 
            inputHandler={this.inputHandler} 
            submitHandler={this.submitHandler}/> 
        </div>
        <div className='comments-container'>
          {this.props.comments.map((comment) => {
            return <Comment 
              body={comment.body}
              author={comment.created_by}
              votes={comment.votes}
              key={comment._id}
              id={comment._id}
              commentVote={this.props.commentVote}/>;
          })}
        </div>
      </div>
    );
  }
}

function mapDispatchToProps (dispatch) {
  return {
    fetchArticle: (id) => {
      dispatch(fetchArticle(id));
    }, 
    fetchComments: (id) => {
      dispatch(fetchComments(id));
    },
    commentVote: (id, vote, comments) => {
      dispatch(commentVote(id, vote, comments));
    },

    addComment: (id, comment) => {
      dispatch(addComment(id, comment));
    }
  };
}

function mapStateToProps (state) {

  return {
    article: state.article, 
    comments: state.comments
  };
}

Article.propTypes = {
  addComment: PropTypes.func.isRequired,
  fetchArticle: PropTypes.func.isRequired, 
  fetchComments: PropTypes.func.isRequired,
  article: PropTypes.object.isRequired,
  title: PropTypes.string.isRequired,
  created_by: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  comments: PropTypes.array.isRequired,
  commentVote: PropTypes.func.isRequired, 
  match: PropTypes.object.isRequired,
  
};


export default connect(mapStateToProps, mapDispatchToProps)(Article);