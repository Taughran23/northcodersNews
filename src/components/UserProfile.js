import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import fetchArticles from '../actions/articles';
import userProfile from '../actions/userProfile';
import articleVote from '../actions/articleVote';
import ArticleCard from './ArticleCard';

import '../css/ArticleList.css';
import '../css/UserProfile.css';

export class UserProfile extends React.Component {
    
  componentDidMount () {
    const userName = this.props.match.params.username;
    this.props.userProfile(userName);
    this.props.fetchArticles();
  }
  render () {
    const userArticles = this.props.articles
      .filter((article) => {
        if (article.created_by === this.props.user.username) {
          return article;
        }
      });
    return (
      <div>
        <section>
          <div className="user-card is-8">
            <div>
              <div>
                <h1 className="title userName"><b>{this.props.user.username}</b></h1>
                <h6 className="subtitle name">{this.props.user.name}</h6>
              </div>
              <div className='avatar-container'>
                <img src={this.props.user.avatar_url}/>
              </div>
            </div>
          </div>
          <div className='ArticleList'>
            {userArticles
              .map(article => 
                <ArticleCard 
                  articleVote={this.props.articleVote}
                  title={article.title} 
                  votes={article.votes} 
                  id={article._id}
                  key={article.title} 
                  user={article.created_by}/>)}
          </div>
        </section>
      </div>
    );
  }
}

function mapDispatchToProps(dispatch) {
  return {
    userProfile: (userName) => {
      dispatch(userProfile(userName));
    }, 
    fetchArticles: () => {
      dispatch(fetchArticles());
    },
    articleVote:(id, vote) => {
      dispatch(articleVote(id, vote));
    }
  };
}

function mapStateToProps (state) {
  return {
    user: state.userProfile, 
    articles: state.articles
  };
}

UserProfile.propTypes = {
  userProfile: PropTypes.func.isRequired,
  fetchArticles: PropTypes.func.isRequired,
  articleVote: PropTypes.func.isRequired, 
  articles: PropTypes.array.isRequired,
  user: PropTypes.object.isRequired,
  match: PropTypes.object.isRequired,
};

export default connect(mapStateToProps, mapDispatchToProps)(UserProfile);