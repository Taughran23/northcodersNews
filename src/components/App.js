import React from 'react';
import { Router, Route } from 'react-router-dom';
import createBrowserHistory from 'history/createBrowserHistory';

import Header from './Header';
import ArticleList from './ArticleList';
import TopicsHeader from './TopicsHeader';
import TopicPage from './TopicPage';
import UserProfile from './UserProfile';
import Article from './Article';
import MobileMenu from './MobileMenu';
import Footer from './Footer';

import '../css/App.css';

const history = createBrowserHistory();

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      active : false
    };
  }
  render () {
    return (
      <Router history={history}>
        <div>
          <Header/>
          <TopicsHeader/>
          <Route exact path='/' component={ArticleList} />
          <Route exact path='/:topic_slug' component={TopicPage} />
          <Route exact path='/users/:username' component={UserProfile} />
          <Route exact path='/articles/:article_id' component={Article} />
          <Route exact path='/menu' component={MobileMenu} />
          <Footer />
        </div>
      </Router>
    );
  }
}

export default App;