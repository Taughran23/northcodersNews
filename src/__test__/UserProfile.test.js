import React from 'react';
import Enzyme , {shallow} from 'enzyme';
import { UserProfile } from '../components/UserProfile';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect';

Enzyme.configure({ adapter: new Adapter() });

describe('<UserProfile />', () => {
    
  it('should render correctly', () => {
    const component = shallow(<UserProfile 
      articles={[]}
      user={{userName:'northcoder'}}
      match={{params:{userName:'northcoder'}}}
      userProfile={() => 'userProfile'}
      fetchArticles={() => [] }/>);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
  it('it renders 1 <UserProfile /> component', () => {
    const component = shallow(<UserProfile 
      articles={[]}
      user={{userName:'northcoder'}}
      match={{params:{userName:'northcoder'}}}
      userProfile={() => 'userProfile'}
      fetchArticles={() => [] }/>);
    expect(component).toHaveLength(1);
  });
});