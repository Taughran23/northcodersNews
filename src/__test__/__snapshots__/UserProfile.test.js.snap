// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserProfile /> should render correctly 1`] = `
<div>
  <section>
    <div
      className="user-card is-8"
    >
      <div>
        <div>
          <h1
            className="title userName"
          >
            <b />
          </h1>
          <h6
            className="subtitle name"
          />
        </div>
        <div
          className="avatar-container"
        >
          <img />
        </div>
      </div>
    </div>
    <div
      id="ArticleList"
    />
  </section>
</div>
`;
